mlflow/pipelines/steps/predict.py

```diff
@@ -107,6 +107,27 @@ def _run(self, output_directory):
                 error_code=BAD_REQUEST,
             )
 
+        # check if output location is already populated as we allow no overwrites
+        output_format = self.step_config["output_format"]
+        output_location = self.step_config["output_location"]
+        if output_format == "parquet" or output_format == "delta":
+            output_populated = os.path.exists(output_location)
+        else:
+            try:
+                output_populated = spark._jsparkSession.catalog().tableExists(output_location)
+            except Exception:
+                # swallow spark failures
+                output_populated = False
+        if output_populated:
+            raise MlflowException(
+                message=(
+                    f"Output location `{output_location}` of format `{output_format}` is already "
+                    "populated and overwrites are not allowed. Please clear or modify the output "
+                    "location and try again."
```

Comment:
Is this behavior really desired? I can clear see for Delta table I just want to keep updating the table with new versions. Not sure about Parquet. Do we have customer feedback on whether to overwrite it?

Reply:
We don't have any customer feedback on this. But I remember preventing overwrites being an earlier decision we made in the Batch Scoring development process. @kriscon-db do you remember the rationale?

Reply:
I could see allowing overwrites for the "table" format only, as Delta Tables keep track of versions. (Not possible for parquet files)

Reply:
I seem to remember someone, at some point saying they preferred not to allow overwrites from one of our standups awhile ago.  Is there someone we should ask to get a final decision on this behavior?  Bilal? XM?

Reply:
I'm inclined to add an option allowing overwrite but default to not allowing it. For Delta tables we should just write with new versions.
Feel free to reach out to XM.

Reply:
Resolving this thread as we've reached a decision on this.